tlair/src/main/scala/at/forsyte/apalache/tla/lir/TlaType1.scala

```diff
@@ -107,9 +107,6 @@ case object StrT1 extends TlaType1 {
  *   unique name of the constant type
  */
 case class ConstT1(name: String) extends TlaType1 {
-  require(ConstT1.isUninterpreted(name) || ConstT1.isAliasReference(name),
-      "ConstT1 accepts identifiers in upper case or $aliasReference, found: " + name)
-
```

Comment:
This was an important invariant about uninterpreted types. Why is it gone now?

Reply:
I noted the reason for this in the commit message, but my understanding is that this is a purely syntactic check, which we deal with in the parser. I don't see why the AST should care how the string looks, and I think this requirement on the AST node constructor is basically an artifact of mixing up levels of abstraction: we require a certain input syntax for parsing out valid type constants -- this is part of valid input to tell us that we need to construct a `ConstT1`. But if we are directly constructing a `ConstT1` the string representation should be irrelevant. The fact that no tests fail in the absence of this (imo, overly strict) requirement seems to support that view. But do you have some additional concern or context to share?

This change is motivated by the fact that it creates conflicts with quint's strings for type constants. We could instead do text munging on the quint strings, but that seems inelegant and to make for an overly tight coupling, when we can instead accommodate via removing the unneeded constraint.

Reply:
This is an internal invariant of the Intermediate Representation. It could be completely different from the syntax, as reflected in the Abstract Syntax Tree (e.g., of TLA+). I don't remember the reason for having this invariant. But I would expect other code depend on it. So instead of removing it, because it is in the way, I would rather try to understand, why it is there and whether anything breaks in Apalache, when this invariant does not hold true.

Reply:
I reverted this change.